docs/abi.rst

```diff
@@ -0,0 +1,560 @@
+:any:`abi.String`                              :code:`string`         Dynamic                           A variable-length byte array assumed to contain UTF-8 encoded content. This is an alias for :code:`abi.DynamicArray[abi.Byte]`.
+:any:`abi.Tuple`\*                             :code:`(...)`          Static if all elements are static A tuple of multiple types
+============================================== ====================== ================================= =======================================================================================================================================================
+
+.. note::
+    \*A proper implementation of :any:`abi.Tuple` requires a variable amount of generic arguments. Python 3.11 will support this with the introduction of `PEP 646 - Variadic Generics <https://peps.python.org/pep-0646/>`_, but until then it will not be possible to make :code:`abi.Tuple` a generic type. As a workaround, we have introduced the following subclasses of :code:`abi.Tuple` for fixed amounts of generic arguments:
+
+    * :any:`abi.Tuple0`: a tuple of zero values, :code:`()`
+    * :any:`abi.Tuple1[T1] <abi.Tuple1>`: a tuple of one value, :code:`(T1)`
+    * :any:`abi.Tuple2[T1,T2] <abi.Tuple2>`: a tuple of two values, :code:`(T1,T2)`
+    * :any:`abi.Tuple3[T1,T2,T3] <abi.Tuple3>`: a tuple of three values, :code:`(T1,T2,T3)`
+    * :any:`abi.Tuple4[T1,T2,T3,T4] <abi.Tuple4>`: a tuple of four values, :code:`(T1,T2,T3,T4)`
+    * :any:`abi.Tuple5[T1,T2,T3,T4,T5] <abi.Tuple5>`: a tuple of five values, :code:`(T1,T2,T3,T4,T5)`
+
+These ARC-4 types are not yet supported in PyTeal:
+
+* Non-power-of-2 unsigned integers under 64 bits, i.e. :code:`uint24`, :code:`uint48`, :code:`uint56`
+* Unsigned integers larger than 64 bits
+* Fixed point unsigned integers, i.e. :code:`ufixed<N>x<M>`
+
+Usage
+"""""""""""""""""""""
+
+Setting Values
+''''''''''''''''
+
+All basic types have a :code:`set()` method which can be used to assign a value. The arguments for this method differ depending on the ABI type. For convenience, here are links to the docs for each class's method:
+
+* :any:`abi.Uint.set()`, which is used by all :code:`abi.Uint` classes and :code:`abi.Byte`
+* :any:`abi.Bool.set()`
+* :any:`abi.StaticArray.set()`
+* :any:`abi.Address.set()`
+* :any:`abi.DynamicArray.set()`
+* :any:`abi.String.set()`
+* :any:`abi.Tuple.set()`
+
+A brief example is below. Please consult the documentation linked above for each method to learn more about specific usage and behavior.
+
+.. code-block:: python
+
+    myAddress = abi.make(abi.Address)
+    myBool = abi.make(abi.Bool)
+    myUint64 = abi.make(abi.Uint64)
+    myTuple = abi.make(abi.Tuple3[abi.Address, abi.Bool, abi.Uint64])
+
+    program = Seq(
+        myAddress.set(Txn.sender()),
+        myBool.set(Txn.fee() == Int(0)),
+        myUint64.set(5000),
```

Comment:
I'm surely wrong because I know nothing and am just pattern matching, but in case I'm not - should this be "Int(5000)"?